x/evm/statedb/store_test.go

```diff
@@ -0,0 +1,315 @@
+package statedb_test
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/evmos/ethermint/x/evm/statedb"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewEphemeralStore_GetRevertKey(t *testing.T) {
+	tests := []struct {
+		name     string
+		maleate  func(store *statedb.EphemeralStore)
+		expected statedb.StoreRevertKey
+	}{
+		{
+			"empty store",
+			func(store *statedb.EphemeralStore) {},
+			statedb.StoreRevertKey{},
+		},
+		{
+			"store with refund",
+			func(store *statedb.EphemeralStore) {
+				store.AddRefund(1)
+			},
+			statedb.StoreRevertKey{
+				RefundIndex:           1,
+				SuicidedAccountsIndex: 0,
+				LogsIndex:             0,
+			},
+		},
+		{
+			"store with suicided account",
+			func(store *statedb.EphemeralStore) {
+				store.SetAccountSuicided(address)
+			},
+			statedb.StoreRevertKey{
+				RefundIndex:           0,
+				SuicidedAccountsIndex: 1,
+				LogsIndex:             0,
+			},
+		},
+		{
+			"store with log",
+			func(store *statedb.EphemeralStore) {
+				store.AddLog(&ethtypes.Log{})
+			},
+			statedb.StoreRevertKey{
+				RefundIndex:           0,
+				SuicidedAccountsIndex: 0,
+				LogsIndex:             1,
+			},
+		},
+		{
+			"store with refund, suicided account and log",
+			func(store *statedb.EphemeralStore) {
+				store.AddRefund(1)
+				store.SetAccountSuicided(address)
+				store.AddLog(&ethtypes.Log{})
+			},
+			statedb.StoreRevertKey{
+				RefundIndex:           1,
+				SuicidedAccountsIndex: 1,
+				LogsIndex:             1,
+			},
+		},
+		{
+			"store with multiple refund, suicided account and log",
+			func(store *statedb.EphemeralStore) {
+				store.AddRefund(1)
+				store.AddRefund(2)
+				store.AddRefund(3)
+				store.SetAccountSuicided(address)
+				store.SetAccountSuicided(address2)
+
+				store.AddLog(&ethtypes.Log{})
+				store.AddLog(&ethtypes.Log{})
+			},
+			statedb.StoreRevertKey{
+				RefundIndex:           3,
+				SuicidedAccountsIndex: 2,
+				LogsIndex:             2,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Not fully sure, how this test works? Looks like `maleate` and `expected` are not used?

Reply:
good catch ty